### PR TITLE
Upgrade most k8s packages to v0.23.4

### PR DESCRIPTION
**Pull request by hpidcock**
Upgrade most k8s packages to v0.23.4

## QA steps

Run k8s tests and bootstrap microk8s, deploy something.

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -232,10 +233,3 @@ replace gopkg.in/yaml.v2 => github.com/juju/yaml v0.0.0-20200420012109-12a32b78d
 replace github.com/dustin/go-humanize v1.0.0 => github.com/dustin/go-humanize v0.0.0-20141228071148-145fabdb1ab7
 
 replace github.com/hashicorp/raft-boltdb => github.com/juju/raft-boltdb v0.0.0-20200518034108-40b112c917c5
-
```
Comment (wallyworld) [inline go.mod:235]: \o/

File: caas/kubernetes/provider/specs/builder.go [hunk 2]
```diff
@@ -52,10 +51,10 @@ func processRawData(data []byte, defaults *apischema.GroupVersionKind, into runt
 	}
 
 	// Make sure the data can decode into ObjectMeta.
-	v := &metadataOnlyObject{}
-	if err = json.CaseSensitiveJsonIterator().Unmarshal(data, v); err != nil {
-		return obj, gvk, errors.Trace(err)
-	}
+	// v := &metadataOnlyObject{}
```
Comment (wallyworld) [inline caas/kubernetes/provider/specs/builder.go:54]: delete me

**Review by wallyworld — APPROVED** [review]

**Review by wallyworld — APPROVED** [review]
Just need to fix static analysis

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$ please

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**